ui/jasmine_test/spec/test.spec.js

```diff
@@ -0,0 +1,20 @@
+import {AudioRecorder} from '../../src/AudioRecorder.js';
+
+describe("Testing the Audio Recorder methods", function() {
+
+    it("Start recording called", function(){
```

Comment:
This doesn't really test the functionality of audioRecorder class. In this case, the class is mostly interacting with the system mediaDevices library, so to test that functionality we can mock this out and verify the system libraries were called.

There is some idea on how to mock these libraries out here:
https://stackoverflow.com/questions/56371404/how-to-accept-permission-for-getusermedia-in-jasmine-testing

Also a good place to look for general guidance on mocking is Jasmine's doc
https://jasmine.github.io/2.0/introduction#section-Spies

What you would be looking to assert for the startRecording is that
 - we get user media (validate w/ spy hasBeenCalled)
 - MediaRecorder is created (validate w/ spy? maybe we provide a way to inject or create an instance of MediaRecorder instead of calling new)
 - mediaRecorder.start is called (we would have the spy from previous bullet, just check hasBeenCalled)
 - what happens if getUserMedia fails, and the `catch` is triggered

What we are actually asserting in this test is basically something like this:

```javascript
let hasBeenCalled = false;

function callTheFunction() {
  hasBeenCalled = true;
}

console.log(hasBeenCalled); // false
callTheFunction();
console.log(hasBeenCalled); // true
```

You can see it would always result in passing tests, no matter what the underlying function is, and it doesn't stress the actual implementation of the function